src/Cli/dotnet/commands/dotnet-workload/WorkloadUtilities.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Deployment.DotNet.Releases;
+
+namespace Microsoft.DotNet.Workloads.Workload
+{
+    internal class WorkloadUtilities
+    {
+        internal static int VersionCompare(string first, string second)
+        {
+            if (first.Equals(second))
```

Comment:
If the versions are strings, what about case sensitivity? Like, `1.0.0.0-preview` vs `1.0.0.0-PREVIEW`? Or can these version number strings not contain anything other than numerics and dots?

Reply:
This first check isn't actually necessary; it's just for performance. If they aren't equal, we start turning parts into versions, and then we use previously built checks. (I think anything after the - is fair game.)